app/questionnaire/questionnaire_schema.py

```diff
@@ -555,14 +559,17 @@ def _update_answer_dependencies_for_list_source(
         but a block depending on the dynamic answers might (such as a calculated summary)
         """
         # Type ignore: section will always exist at this point, same with optional returns below
-        section: ImmutableDict = self.get_section_for_block_id(block_id)  # type: ignore
-        list_collector: ImmutableDict = self.get_list_collector_for_list(  # type: ignore
-            section=section,
+        section_id = self._list_name_to_section_id_origin_map[list_name]
+
+        section = self.get_section(section_id)
+
+        list_collector = self.get_list_collector_for_list(
```

Comment:
Non blocking
Since you now have this map - could `get_list_collector_for_list` be improved by including section lookup instead of it needing to be looked up and passed in everywhere its being used?